### PR TITLE
Parallelize Accumulo index range scanning

### PR DESCRIPTION
**Pull request by adamjshook**
This adds an ExecutorService to the IndexLookup class, submitting all
tasks to be run in parallel and waiting for the results to come back
for each column constraint.

### CODE REVIEW

**Comment by adamjshook** [comment]
Bump 😄 Would it be possible to take a look at this sometime soon?  Thank you!

File: presto-accumulo/src/main/java/com/facebook/presto/accumulo/index/IndexLookup.java [hunk 1]
```diff
@@ -297,55 +318,58 @@ private long getNumRowsInTable(String metricsTable, Authorizations auths)
...
+                scan.close();
+                return columnRanges;
+            }));
+        }
 
-            // If finalRanges is null, we have not yet added any column ranges
-            if (finalRanges == null) {
-                finalRanges = new HashSet<>();
-                finalRanges.addAll(columnRanges);
+        tasks.forEach(future ->
```
Comment (electrum) [inline presto-accumulo/src/main/java/com/facebook/presto/accumulo/index/IndexLookup.java]: Formatting nit: this should be on previous line

File: presto-accumulo/src/main/java/com/facebook/presto/accumulo/index/IndexLookup.java [hunk 2]
```diff
@@ -297,55 +318,58 @@ private long getNumRowsInTable(String metricsTable, Authorizations auths)
...
+                    // This is your set intersection operation!
+                    finalRanges.retainAll(future.get());
+                }
             }
-            else {
-                // Retain only the row IDs for this column that have already been added
-                // This is your set intersection operation!
-                finalRanges.retainAll(columnRanges);
+            catch (ExecutionException | InterruptedException e) {
+                throw new PrestoException(UNEXPECTED_ACCUMULO_ERROR, "Exception when getting index ranges", e);
```
Comment (electrum) [inline presto-accumulo/src/main/java/com/facebook/presto/accumulo/index/IndexLookup.java]: After catching `InterruptedException`, we should restore the interrupted status before throwing

File: presto-accumulo/src/main/java/com/facebook/presto/accumulo/index/IndexLookup.java [hunk 3]
```diff
@@ -297,55 +318,58 @@ private long getNumRowsInTable(String metricsTable, Authorizations auths)
...
+
+                    // Add to our column ranges if it is in one of the row ID ranges
+                    if (inRange(tmpQualifier, rowIDRanges)) {
+                        columnRanges.add(new Range(tmpQualifier));
+                    }
                 }
-            }
 
-            LOG.debug("Retrieved %d ranges for column %s", columnRanges.size(), constraintEntry.getKey().getName());
+                LOG.info("Retrieved %d ranges for index column %s", columnRanges.size(), constraintEntry.getKey().getName());
```
Comment (electrum) [inline presto-accumulo/src/main/java/com/facebook/presto/accumulo/index/IndexLookup.java]: Did you mean to change the log level? Having this at info level seems rather spammy (we avoid excessive default logging).

**Comment by electrum** [comment]
Merged, thanks!

**Pull request closed without merging**